eppo/src/main/java/cloud/eppo/android/FlagEvaluator.java

```diff
@@ -0,0 +1,160 @@
+        evaluationResult.setExtraLogging(extraLogging);
+        evaluationResult.setDoLog(doLog);
+
+        if (isConfigObfuscated) {
+            // Need to unobfuscate the evaluation result
+            evaluationResult.setFlagKey(flagKey);
+            evaluationResult.setAllocationKey(base64Decode(allocationKey));
+            if (variation != null) {
+                Variation decodedVariation = new Variation();
+                decodedVariation.setKey(base64Decode(variation.getKey()));
+                EppoValue decodedValue = EppoValue.nullValue();
+                if (!variation.getValue().isNull()) {
+                    String stringValue = base64Decode(variation.getValue().stringValue());
+                    switch (flag.getVariationType()) {
+                        case BOOLEAN:
+                            decodedValue = EppoValue.valueOf("true".equals(stringValue));
+                            break;
+                        case INTEGER:
+                        case NUMERIC:
+                            decodedValue = EppoValue.valueOf(Double.parseDouble(stringValue));
+                            break;
+                        case STRING:
+                        case JSON:
+                            decodedValue = EppoValue.valueOf(stringValue);
+                            break;
+                        default:
+                            throw new UnsupportedOperationException("Unexpected variation type for decoding obfuscated variation: " + flag.getVariationType());
+                    }
+                }
+                decodedVariation.setValue(decodedValue);
+                evaluationResult.setVariation(decodedVariation);
+            }
+        }
+
+        return evaluationResult;
+    }
+
+    private static boolean allShardsMatch(Split split, String subjectKey, int totalShards, boolean isObfuscated) {
+        if (split.getShards() == null || split.getShards().isEmpty()) {
+            // Default to matching if no explicit shards
+            return true;
+        }
+
+        boolean allShardsMatch = true;
+        for (Shard shard : split.getShards()) {
+            if (!matchesShard(shard, subjectKey, totalShards, isObfuscated)) {
+                allShardsMatch = false;
+                break;
+            }
+        }
+        return allShardsMatch;
+    }
+
+    private static boolean matchesShard(Shard shard, String subjectKey, int totalShards, boolean isObfuscated) {
+        String salt = shard.getSalt();
+        if (isObfuscated) {
+            salt = base64Decode(salt);
+        }
+        String hashKey = salt+"-"+subjectKey;
+        int assignedShard = Utils.getShard(hashKey, totalShards);
+        boolean inRange = false;
+        for (Range range : shard.getRanges()) {
+            if (assignedShard >= range.getStart() && assignedShard < range.getEnd()) {
+                inRange = true;
+                break;
+            }
+        }
+        return inRange;
```

Comment:
same:
```suggestion
        for (Range range : shard.getRanges()) {
            if (assignedShard >= range.getStart() && assignedShard < range.getEnd()) {
                return true;
            }
        }
        return false;
```

Reply:
This we need as-is as I think if there is no range, then we consider it always in range. Or I could do this with a guard clause up top. Allthough all other things equal, I have a slight bias towards fewer `return` statements.

The other thing with shards was probably mirroring this structure.

Reply:
Correction: for this, if there is no range, it's not in range, so changing per your suggestion, and it will have the same structure as the other change you suggested, which I like the parallelism.